users/views.py

```diff
@@ -1,3 +1,33 @@
-from django.shortcuts import render
+import bcrypt
+import jwt
 
-# Create your views here.
+from datetime               import datetime, timedelta
+from django.http            import JsonResponse
+from django.views           import View
+from users.models           import User
+from my_settings            import ALGORITHM, SECRET_KEY
+
+class SignInView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+
+        try:
+            email          = data['email']
+            user           = User.objects.get(email=email)
+            user_saved_pwd = user.password
+            password       = data['password']
+
+            if not bcrypt.checkpw(password.encode('utf-8'), user_saved_pwd.encode('utf-8')):
+
+                current_time          = datetime.utcnow()
+                expiration_time       = timedelta(weeks=3)
+                token_expiration_time = current_time+expiration_time
+                jwt_access_token      = jwt.encode({'id':user.id, 'exp':token_expiration_time}, SECRET_KEY, ALGORITHM)
```

Comment:
모든 값들을 전부 변수에 꼭 안 담아도 됩니다.

```suggestion
                payload = {
                    'id'  : user.id, 
                    'exp' : datetime.utcnow() + timedelta(weeks=3)
                }
                jwt_access_token = jwt.encode(payload, SECRET_KEY, ALGORITHM)
```